### ISSUE TITLE
schematic review

### ISSUE THREAD

**Issue by hartytp** [sinara-hw/Kasli#53]
@marmeladapk I looked at the schematics. They look good!

I've raised a few issues, mainly with questions. Other that that LGTM. NB however I haven't done the kind of thorough review I've done for the EEMs (checking every pin on every IC, reference designs, etc), I assume that wasn't expected for this release.

@sbourdeauducq @WeiDaZhang can you sign off on the WR parts?

@jordens @sbourdeauducq are you happy to sign off on the design over all?

**Comment by gkasprow** [comment]
- [ ] is there any reason to keep the SATA connector? 
- [ ] what about removing boot mode selection resistors? Are there any plans to change it?
- [ ] the same with DXN, DXP resistors
- [ ] I'd replace L23, L35 with 1..2.2u inductors. FB has near 0R impedance at the buck frequency. Don't forget about adding poor tantalium to dump LC oscillations.
- [ ] I2C_3V3_SDA/SCL should be called I2C_3V3_SW_SDA/SCL
- [ ] add ordinary diode in parallel to J21. This is to protect the MOSFET and fan
- [ ] 0.5pF caps are not needed with XFMR_MINI-CIRCUITS_TC2-1TX+
- [ ] connect pin 6 of TC2-1TX+ to SMA GND via DNP 0R. At the moment we cannot use isolation potential of the transformer.

**Comment by sbourdeauducq** [comment]
>     * is there any reason to keep the SATA connector?

We use it for connecting two Kasli in the same crate with DRTIO. It's a somewhat uncommon use case though.

**Comment by gkasprow** [comment]
Now you can do the same with sfp patchcord

**Comment by hartytp** [comment]
@jordens @sbourdeauducq have you signed off on this design review?

**Comment by jordens** [comment]
Oh. And excellent work on the git commit messages, changelog, and issue tracking @marmeladapk ! Thanks.

**Comment by hartytp** [comment]
> Does it make sense to go for the Si53340-B-GM instead of ADCLK948 IC1 and add two more MMCX clock outs? We use it on Urukul already. And it can run at 2.5 V which may make ditching the Si5324 together with its supply more efficient later.

Fine by me. @WeiDaZhang any objections.

@WeiDaZhang please can you confirm that you're happy to sign off on the clocking part of this design asap (apologies if you've already done that!).

Other than that, I'm happy to sign off

**Comment by hartytp** [comment]
@WeiDaZhang did you check the rest of the clocking? (I think you did this ages ago, but want to double check you still happy) If so, we just need @sbourdeauducq to sign off on the WR PLL implementation from his side and then I think all reviews are complete.

Would be great to get this off to manufacture in the next week or so

**Comment by WeiDaZhang** [comment]
> Fine by me. @WeiDaZhang any objections.

We measured before that ADCLK948 has flicker phase noise of -145dBc/Hz@10kHz offset at 1GHz carrier.
Si53340-B-GM seems to be -137dBc/Hz@10kHz offset at 312MHz, which is slightly worse.

Our WR is no better than 1x10^-13 mod Allan, which about -120dBc/Hz@10kHz offset at 312MHz.
So, should not make any difference.

**Comment by hartytp** [comment]
Yes. This is the kind of thing that would have worried me a year ago, but I think we now understand the performance bottlenecks well enough to know that this will not be a limitation (eg compared to the transceiver pll)

@WeiDaZhang are you happy to sign off on the complete design?

**Comment by WeiDaZhang** [comment]
> Yes. This is the kind of thing that would have worried me a year ago, but I think we now understand the performance bottlenecks well enough to know that this will not be a limitation (eg compared to the transceiver pll)
> 
> @WeiDaZhang are you happy to sign off on the complete design?

Happy

**Comment by hartytp** [comment]
Cool so we just need @sbourdeauducq to sign off on the wr connections and then the review is complete

**Comment by sbourdeauducq** [comment]
Except for the support for CMOS Si549 oscillators, this looks OK to me.

**Comment by hartytp** [comment]
@marmeladapk Okay, everyone has now completed their reviews. So...when can this go to manufacture?

**Comment by hartytp** [comment]
@marmeladapk heads up: the grant we used to pay for our creotech kaslis expires soon so we need to get them to manufacture ASAP to avoid problems

**Comment by hartytp** [comment]
Ping @marmeladapk @gkasprow we really need to get kasli 2.0 sent to manufacture in the next week before the grant that paid for the work ends. This is very important to us. In any case, it seems like almost all the work is done already

**Comment by marmeladapk** [comment]
@hartytp Layout and schematics are finished.

**Comment by hartytp** [comment]
Nice!

I've had a skim over the schematics and it looks good to me. I haven't done the usual thorough design review (check every pin description, eval board/ref design, pin abs max ratings, required capacitances, etc etc) as I'm assuming you did that @marmeladapk 

@WeiDaZhang can you sign off on the WR/clocking please?

@sbourdeauducq @jordens are you happy with this revision? I'd like to send boards to manufacture on monday if possible.

**Comment by hartytp** [comment]
@marmeladapk can you remind me which areas had changes? It's worth us carefully reviewing all changes to ensure we don't break anything...

**Comment by hartytp** [comment]
I went over the clock page one last time (the only page that has significant changes that I feel qualified to review). Other than cosmetics I don't see any issues.

Any objections to cutting this release on Monday morning?

**Comment by WeiDaZhang** [comment]
> I went over the clock page one last time (the only page that has significant changes that I feel qualified to review). Other than cosmetics I don't see any issues.
> 
> Any objections to cutting this release on Monday morning?

May I ask why that?
![image](https://user-images.githubusercontent.com/22749113/69885893-72b5b100-12d7-11ea-8b6b-316f3247ea3a.png)
Select the power between 2.5/3.3 I suppose.
If so, shall we use P3V3_LDO1 instead then, as Si549s shall not be used at the same time when Si5324 is in use.

**Comment by hartytp** [comment]
Thanks @marmeladapk!

I'm assuming everyone has completed their reviews (@WeiDaZhang @sbourdeauducq @jordens ) so we can close this and go to manufacture on Monday once the last straggling issues are closed.

**Comment by hartytp** [comment]
I saw a box labelled Kasli 2 at CTI so I guess this exists now...

**Comment by hartytp** [comment]
@marmeladapk we really need these in our lab ASAP. When can you ship them?

**Comment by marmeladapk** [comment]
@hartytp I don't even have them on my desk. Do you really want us to send you untested hardware without panels? AFAIK we should receive Kaslis on Monday from production (it's currently during inspection).

Besides you should write to Anna about this not to me, as it's her decision not mine when to ship it.

**Comment by hartytp** [comment]
Thanks for the update. Please do test before shipping. Okay, so if receive them on Monday, is it realistic to think that testing will be complete by end of next week?

**Comment by akaminska** [comment]
Hi Tom @hartytp , Kasli is still in optical inspection after production. We should get the boards for testing Tuesday morning. We will keep you posted on the progress.

**Comment by hartytp** [comment]
Great! Thanks for the update.

**Comment by marmeladapk** [comment]
@hartytp @jordens 

![IMG_20200304_110615__01](https://user-images.githubusercontent.com/24512604/75871437-da1a5a00-5e0c-11ea-8268-d191c02f6e77.jpg)

![IMG_20200304_113328__01](https://user-images.githubusercontent.com/24512604/75871092-4fd1f600-5e0c-11ea-90b4-3ffc40028623.jpg)

This SFP cage is massive.

We're now validating if our tests work correctly with the new hardware.

**Comment by sbourdeauducq** [comment]
Looks good!
Where is the fan plugged?

**Comment by marmeladapk** [comment]
J21, above the FPGA. (2-pin header). Currently the pins are too short, but I plan to change them to header with key, which won't allow reverse connection.

**Comment by hartytp** [comment]
Awesome, looks beautiful! Good job @marmeladapk

**Comment by hartytp** [comment]
When do the panels arrive?

**Comment by akaminska** [comment]
On Monday at the latest.

**Comment by jordens** [comment]
Re availability of the SFP cages: I noticed that even Cisco would use two 2x1 cages next to each other when they need 4 SFP and not a 2x2.
And what are we going to do with all the unused space when we go to cPCIs...

**Comment by gkasprow** [comment]
CPCIS is not an issue because Kasli would be rotated 180degs and the panel can still be 8HP

**Comment by jordens** [comment]
I'm not worried about panel space (and for 4 HP, the cooling stack may or may not be in the way as well).
I was wondering whether there is anything useful we can do with all the unused space of the IDC rats nest which occupies half the entire board area.

**Comment by marmeladapk** [comment]
Most of that space will still be required to escape from under the bga package. But perhaps we could make Kasli and other CPCIS boards shorter (160mm)?

**Comment by gkasprow** [comment]
The idea behind existing Kasli was to add simple adapter that converts EEMs to CPCI.

**Comment by jordens** [comment]
I think it will be much better to draw a line and transition a sufficient set of several modules to cPCIs in one go without adapters.

**Comment by gkasprow** [comment]
Yes, but for the moment essential part of the community seems to be happy with existing solution

**Comment by jordens** [comment]
The community that M-Labs and QUARTIQ serve would very much appreciate a better solution.

**Comment by marmeladapk** [comment]
Kasli v1.1 gateware with uart pins moved to new positions.
```
Initializing SDRAM...
Read leveling scan:
Module 1:
00000000000111111111110000000000
Module 0:
00000000000111111111110000000000
Read leveling: 16+-5 16+-5 done
SDRAM initialized
Memory test passed

Booting from flash...
Starting firmware.
[     0.000004s]  INFO(satman): ARTIQ satellite manager starting...
[     0.005612s]  INFO(satman): software ident 5.0.dev+848.g2a909839.dirty;satellite
[     0.013025s]  INFO(satman): gateware ident 5.0.dev+848.g2a909839.dirty;satellite
panic at src/libcore/result.rs:945:5: cannot initialize Si5324: "Si5324 failed to ack write address"
```
I also moved I2C SDA. It seems that I2C works at least initially (no I2C switch error). We'll test this further using our test bench.

**Comment by akaminska** [comment]
From the point of view of managing hardware production and stocks, focusing on a specific format of cards for the future instead of having two versions of most cards would make life easier

**Comment by hartytp** [comment]
Awesome @marmeladapk 

Re cpics: I can see this being what we want long term but would be nice to have a play with the adapter boards and existing eems before making a decision

**Comment by gkasprow** [comment]
this is exactly what we do right now. 3 adapters would be needed to serve all EEM boards.
